partner_vat_unique/models/res_partner.py

```diff
@@ -0,0 +1,24 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Grant Thornton Spain - Ismael Calvo <ismael.calvo@es.gt.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, models, _
+from odoo.exceptions import ValidationError
+
+
+class ResPartner(models.Model):
+    _inherit = 'res.partner'
+
+    @api.constrains('vat')
+    def _check_vat_unique(self):
+        for record in self:
+            if not record.parent_id:
+                results = self.env['res.partner'].read_group([
+                    ('parent_id', '=', False),
+                    ('vat', '=', record.vat),
+                    ('id', '!=', record.id)
+                ], ['vat'], ['vat'])
+            for result in results:
```

Comment:
This `for` makes no sense given what it does inside (and what I commented above). Replace by an `if`.